src/realtime-messaging/realtime-messaging.gateway.ts

```diff
@@ -0,0 +1,83 @@
+import { Injectable } from "@nestjs/common";
+import { SubscribeMessage, WebSocketGateway, WebSocketServer } from "@nestjs/websockets";
+
+import { Server, Socket } from "socket.io";
+
+import { UserRepository } from "@/users/users.repository";
+
+import { IMessagePayload } from "./realtime-message.interface";
+
+@Injectable()
+@WebSocketGateway(Number(process.env.BE_WS_PORT), {
+  cors: {
+    origin: "*",
+  },
+})
+export class RealtimeMessagingGateway {
+  @WebSocketServer()
+  server!: Server;
+
+  constructor(private userRepository: UserRepository) {}
+
+  @SubscribeMessage("sendMessage")
+  async handleMessage(client: Socket, payload: IMessagePayload) {
+    const { classroomId, content, userId } = payload;
+    const sender = await this.userRepository.findOne(userId);
+
+    if (!sender) {
+      client.emit("error", "User not found");
+      return;
+    }
+
+    this.server.to(`classroom-${classroomId}`).emit("receiveMessage", {
+      content,
+      sender,
+      classroomId,
+      timestamp: new Date(),
+    });
+  }
+
+  @SubscribeMessage("joinClassroom")
+  async handleJoinClassroom(client: Socket, payload: { classroomId: number; userId: number }) {
+    const { classroomId, userId } = payload;
+    const user = await this.userRepository.findOne(userId);
+
+    if (!user) {
+      client.emit("error", "User not found");
+      return;
+    }
+
+    client.join(`classroom-${classroomId}`);
+    const userName = `${user.firstName} ${user.lastName}`;
+
+    this.server.to(`classroom-${classroomId}`).emit("userJoined", {
+      userId: user.id,
+      userName,
+      userType: user.userType,
+      classroomId,
+      timestamp: new Date(),
+    });
```

Comment:
Same comment. Do we need to emit this?

Reply:
No, it was just for testing purpose. I have removed it.